cardano-db-sync/src/Cardano/DbSync/Config.hs

```diff
@@ -96,10 +96,9 @@ mkAdjustPath :: SyncPreConfig -> (FilePath -> FilePath)
 mkAdjustPath cfg fp = takeDirectory (pcNodeConfigFilePath cfg) </> fp
 
 -- check both whitelist but also checking plutus Maybes first
--- TODO: cmdv: unsure if this is correct because if plutusMaybeCheck fails then no multiasset whitelist is not checked
 plutusMultiAssetWhitelistCheck :: SyncEnv -> [Generic.TxOut] -> Bool
 plutusMultiAssetWhitelistCheck syncEnv txOuts =
-  plutusMaybeCheck txOuts && (plutusWhitelistCheck syncEnv txOuts || multiAssetWhitelistCheck syncEnv txOuts)
+  plutusMaybeCheck txOuts || (plutusWhitelistCheck syncEnv txOuts || multiAssetWhitelistCheck syncEnv txOuts)
```

Comment:
`plutusWhitelistCheck` is called in `plutusMaybeCheck`, so why is it also called here?

Reply:
I can't see `plutusWhitelistCheck` inside `plutusMaybeCheck`, am I understanding you right?

Reply:
Oh sorry I got confused by identation. 

Actually I don't see what's the purpose of `plutusMaybeCheck`. It may return `False` even with `PlutusEnable`, which should never happen. `plutusWhitelistCheck` already does the necessary work.

Reply:
The was my original comment
> -- TODO: cmdv: unsure if this is correct because if plutusMaybeCheck fails then no multiasset whitelist is not checked

I was trying to make what we spoke about and push the checks as far up the chain and I may have miss understood what you meant and that's why I left this comment there about the `plutusMaybeCheck`. It's there as a check so you don't do any whitelist checks if there isn't any `txOutScript` or `txOutAddress` and that's what it was an `&&` not an `||` so this is:

```
if plutusMaybeCheck txOuts
    then (plutusWhitelistCheck syncEnv txOuts || multiAssetWhitelistCheck syncEnv txOuts)
    else False
```

which is essentially bellow as it will return false straight away if `plutusMaybeCheck txOuts` if false:

```
plutusMaybeCheck txOuts && (plutusWhitelistCheck syncEnv txOuts || multiAssetWhitelistCheck syncEnv txOuts)
```

Reply:
I still can't fully understand what does `plutusMaybeCheck` check, since all necessary checks are done in `plutusWhitelistCheck`. Using `or` instead of `and` here will simply give `True` in more cases than we want to.

Reply:
it was just to save doing big `elem` checks if they're not needed. But it's all good we can just do it inside the whitelist checks.